### DISCUSSION TITLE
A Universal Windows port of Leaf

### DISCUSSION THREAD

**Discussion by bdbai** [eycorsican/leaf#92] [Show and tell]
[Maple](https://github.com/YtFlow/Maple) is a Universal Windows app that adapts Leaf to the UWP VPN Platform. It acts as a VPN plugin, just like how iLeaf works on iOS. There is also a UI for configuration editing.

Theoretically this app should work on with Windows 10 Mobile. However some of our outdated dependency crates (specifically `winapi 0.2.8`) still lack ARM support which was added in a later version. 😌
Is there a plan to migrate them?

**Comment by eycorsican** [comment]
A Windows VPN without dependency on any third-party TUN/TAP drivers, THIS IS AWESOME!

**Comment by eycorsican** [comment]
> Theoretically this app should work on with Windows 10 Mobile. However some of our outdated dependency crates (specifically winapi 0.2.8) still lack ARM support which was added in a later version. 😌
Is there a plan to migrate them?

Just have a look at our dependencies, we don't directly depend on `winapi`. It seems we have indirect dependency via tokio, I believe we should migrate to tokio 1.0 first, I'll start the migration process in the coming days.
